tooling/src/main/java/org/opencds/cqf/tooling/operations/stripcontent/BaseContentStripper.java

```diff
@@ -0,0 +1,198 @@
+package org.opencds.cqf.tooling.operations.stripcontent;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import org.hl7.fhir.instance.model.api.IAnyResource;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r5.model.Attachment;
+import org.hl7.fhir.r5.model.DomainResource;
+import org.hl7.fhir.r5.model.Extension;
+import org.hl7.fhir.r5.model.Library;
+import org.hl7.fhir.r5.model.Measure;
+import org.hl7.fhir.r5.model.Parameters;
+import org.hl7.fhir.r5.model.PlanDefinition;
+import org.hl7.fhir.r5.model.Questionnaire;
+import org.hl7.fhir.r5.model.RelatedArtifact;
+import org.hl7.fhir.r5.model.Resource;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.parser.DataFormatException;
+import ca.uhn.fhir.parser.IParser;
+
+/**
+ * This class is used to strip autogenerated content from FHIR resources. This includes narrative,
+ * extensions added by the tooling, related artifacts that are auto detected from the CQL,
+ * contained resources added by the tooling, and ELM generated from the CQL.
+ * 
+ * This class converts the resource to its R5 equivalent, strips the content, and then converts
+ * back to the original FHIR version.
+ * 
+ * The T parameter is used to specify the version of the Resource base class to use for the operation
+ * and conversions.
+ */
+abstract class BaseContentStripper<T extends IAnyResource> implements ContentStripper {
+    protected abstract FhirContext context();
+    protected abstract Resource convertToR5(T resource);
+    protected abstract T convertFromR5(Resource resource);
+
+    @SuppressWarnings("unchecked")
+    public void stripFile(File inputFile, File outputFile, ContentStripperOptions options) {
+        var resource = parseResource(inputFile);
+        var upgraded = convertToR5((T)resource);
+        stripResource(upgraded, outputFile, options);
+        var downgraded = convertFromR5(upgraded);
+        writeResource(outputFile, downgraded);
+    }
+
+    protected void writeContent(File f, String content) {
+        if (!f.getParentFile().exists()) {
+            f.getParentFile().mkdirs();
+        }
+
+        try (var writer = new BufferedWriter(new FileWriter(f))) {
+            writer.write(content);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected IParser parserForFile(File file) {
+        if (file.getName().endsWith(".json")) {
+            return context().newJsonParser();
+        } else if (file.getName().endsWith(".xml")) {
+            return context().newXmlParser();
+        } else {
+            throw new IllegalArgumentException(String.format("unsupported file type: %s", file.getName()));
+        }
+    }
+
+    protected IBaseResource parseResource(File file) {
```

Comment:
IOUtils utility functions should be available for you to reuse here as well.

Reply:
Same as previous.